format-delimited/src/main/java/io/cdap/plugin/format/delimited/common/DataTypeDetectorStatusKeeper.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.delimited.common;
+
+import io.cdap.cdap.api.data.schema.Schema;
+
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Aggregates and arranges all data types met while investigating a data file for automated data type detection. Has
+ * functionalities to detect the data type of a single value or a single column.
+ */
+public class DataTypeDetectorStatusKeeper {
+
+  private Map<String, EnumSet<DataType>> dataTypeDetectionStatus = new LinkedHashMap<>();
+
+  /**
+   * Detects the datatype of a single value.
+   *
+   * @param value Value whom the data type would be detected from.
+   * @return The {@link DataType} of the given value.
+   */
+  public static DataType detectsValueDataType(String value) {
+    return TypeInference.getDataType(value);
+  }
+
+  /**
+   * Detects the data type of a single column.
+   *
+   * @param dataTypes Data types met while investigating the a column for automated data type detection.
+   * @return The {@link Schema} of the given column.
+   */
+  public static Schema detectColumnDataType(EnumSet<DataType> dataTypes) {
+    DataType dataType = dataTypes.stream().max(Comparator.comparing(v -> v.priority)).orElse(DataType.STRING);
+    boolean isNull = dataTypes.stream().anyMatch(v -> v.name.equalsIgnoreCase(DataType.EMPTY.name));
+
+    switch (dataType) {
+      case INTEGER:
+        return getSchemaOfType(Schema.Type.INT, isNull);
+      case DOUBLE:
+        return getSchemaOfType(Schema.Type.DOUBLE, isNull);
+      case LONG:
+        return getSchemaOfType(Schema.Type.LONG, isNull);
+      case DATE:
+        return getSchemaOfType(Schema.Type.STRING, isNull);
```

Comment:
Let's add a comment indicating why date and time types are detected as string (PLUGIN-637).